Dockerfile

```diff
@@ -1,24 +1,30 @@
-FROM centos:8 as base
+FROM quay.io/centos/centos:stream9 as base
 ARG channel="stable"
 ARG location
 
 RUN [ -z "${channel}" ] && echo "ARG channel is required" && exit 1 || true
 
-RUN yum -y install jq
-RUN curl https://builds.coreos.fedoraproject.org/streams/${channel}.json -o stable.json && \
-	cat stable.json | jq '.architectures.x86_64.artifacts.qemu.release' | tr -d '"'
+RUN yum -y install jq xz
+RUN ARCH=$(uname -m) ; echo $ARCH \
+    ; if [ "$ARCH" == "s390x" ] ; then ARCH=s390x ; elif [ "$ARCH" == "x86_64" ] ; then ARCH=x86_64 ; fi \
+    ; curl https://builds.coreos.fedoraproject.org/streams/${channel}.json -o stable.json && \
+        cat stable.json | jq --arg arch "$ARCH" '.architectures[$arch].artifacts.qemu.release' | tr -d '"' 
+
 
 
 FROM base AS executor-img
 
 RUN if [[ -z "$arg" ]] ; then \
-	echo "Downloading" $(cat stable.json | jq '.architectures.x86_64.artifacts.qemu.formats."qcow2.xz".disk.location' | tr -d '"') && \
-	curl -s -o coreos_production_qemu_image.qcow2.xz $(cat stable.json | jq '.architectures.x86_64.artifacts.qemu.formats."qcow2.xz".disk.location' | tr -d '"') && unxz coreos_production_qemu_image.qcow2.xz ; \
-	else \
+	ARCH=$(uname -m) ; echo $ARCH ; \
+	if [ "$ARCH" == "s390x" ] ; then ARCH=s390x ; elif [ "$ARCH" == "x86_64" ] ; then ARCH=x86_64 ; fi ; \
+	echo "Downloading" $(cat stable.json | jq --arg arch "$ARCH" '.architectures[$arch].artifacts.qemu.formats."qcow2.xz".disk.location' | tr -d '"') && \
+        curl -s -o coreos_production_qemu_image.qcow2.xz $(cat stable.json | jq --arg arch "$ARCH" '.architectures[$arch].artifacts.qemu.formats."qcow2.xz".disk.location' | tr -d '"') && \
+        unxz coreos_production_qemu_image.qcow2.xz ; \
+    else \
 	echo "Downloading" ${location} && \
-	curl -s -o coreos_production_qemu_image.qcow2.xz ${location} && unxz coreos_production_qemu_image.qcow2.xz \
-	; fi
-
+        curl -s -o coreos_production_qemu_image.qcow2.xz ${location} && \
```

Comment:
You mix tabs and spaces. Please use either tabs or spaces.

Reply:
Will change and update it properly.

Thanks